src/Components/test/testassets/BasicTestApp/SignalRClientComponent.razor

```diff
@@ -47,6 +50,12 @@
         await hubConnection.SendAsync("SendMessage", "SignalR Client", $"Echo {transportType}");
     }
 
+    protected async Task GetUserAgentHeader()
+    {
+        userAgent = await hubConnection.InvokeAsync<string>("GetHeaderValue", "X-SignalR-User-Agent");
```

Comment:
Seems like you want both user agents right? To make sure the normal is set as well.

Reply:
Normal is set by the browser, we shouldn't be testing browser behavior

Reply:
So before our code was just ineffective

Reply:
In WASM it was ineffective in some browsers, and set in others
Details at https://github.com/dotnet/aspnetcore/issues/29183#issue-782700972